activesupport/test/core_ext/range_ext_test.rb

```diff
@@ -32,6 +32,26 @@ def test_date_range
     assert_instance_of Range, DateTime.new..DateTime::Infinity.new
   end
 
+  def test_date_range_includes_date
+    range = Date.today.beginning_of_month..Date.today.end_of_month
+    assert(range.include?(Date.today))
+  end
+
+  def test_date_range_includes_date_time
+    range = Date.today.beginning_of_month..Date.today.end_of_month
+    assert(range.include?(DateTime.now))
+  end
+
+  def test_date_time_range_includes_date
+    range = DateTime.now.beginning_of_month..DateTime.now.end_of_month
+    assert(range.include?(Date.today))
+  end
+
+  def test_date_time_range_includes_date_time
+    range = DateTime.now.beginning_of_month..DateTime.now.end_of_month
+    assert(range.include?(DateTime.now))
+  end
+
```

Comment:
Should also add test cases for `Time` inclusion. Like:

```ruby
(Date.yesterday..Date.tomorrow).include?(Time.now)
(Date.yesterday..Date.tomorrow).include?(Time.zone.now)
```

Reply:
Added tests for this.

Reply:
Added tests for these scenarios, too.